### ISSUE TITLE
build error app-debug.apk does not exist under flutter master version

### ISSUE THREAD

**Issue by stoneLee81** [flutter/flutter#116173]
When I upgrade flutter stable 3.3.9 to master version, and build debugging APK for testing, When compiling debugging android APK, prompt '"build/host/outputs/apk/app-debug.apk" does not exist. error message, and build failed. Has anyone encountered this kind of problem?

**Comment by huycozy** [comment]
Hi @stoneLee81 
When opening an issue, please try to provide all or some of the information indicated in the template so that we can address this issue properly.

<details>
<summary>Issue Template</summary>

```
If you are looking for support, please check out our documentation
or consider asking a question on Stack Overflow:
- https://flutter.dev/
- https://api.flutter.dev/
- https://stackoverflow.com/questions/tagged/flutter?sort=frequent

If you have found a bug or if our documentation doesn't have an answer
to what you're looking for, then fill our the template below. Please read
our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports

Steps to Reproduce

Please tell us exactly how to reproduce the problem you are running into.

Please attach a small application (ideally just one main.dart file) that
reproduces the problem. You could use https://gist.github.com/ for this.

If the problem is with your application's rendering, then please attach
a screenshot and explain what the problem is.

1. ...
2. ...
3. ...


Please tell us which target platform(s) the problem occurs (Android / iOS / Web / macOS / Linux / Windows)
Which target OS version, for Web, browser, is the test system running?
Does the problem occur on emulator/simulator as well as on physical devices?

Target Platform:
Target OS version/browser:
Devices:

Run your application with `flutter run --verbose` and attach all the
log output below between the lines with the backticks. If there is an
exception, please see if the error message includes enough information
to explain how to solve the issue.

Run `flutter analyze` and attach any output of that command below.
If there are any analysis errors, try resolving them before filing this issue.

Finally, paste the output of running `flutter doctor -v` here.
```
</details>

To be more precise, please provide the information indicated below:
- The output of `flutter doctor -v`, `flutter run -v`. If the log is too long, please save it into a file and attach it here.
- The affected device information (model, OS version)

Thank you!

**Comment by stoneLee81** [comment]
![Snip20221130_13](https://user-images.githubusercontent.com/18001708/204754970-4def257f-d595-40d4-a50c-cc820c645b3e.png)

_flutter run -v log：_
[flutter.run.3.6.0-12.log](https://github.com/flutter/flutter/files/10121509/flutter.run.3.6.0-12.log)

_flutter doctor_
`[!] Flutter (Channel master, 3.6.0-12.0.pre.20, on Mac OS X 10.15.7 19H15 darwin-x64, locale en-CN)
    ! Warning: `dart` on your path resolves to /Users/haowmazs/data/software/dart/dart-sdk/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/haowmazs/data/software/flutter. Consider adding /Users/haowmazs/data/software/flutter/bin to the front of your path.
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.1)
[✓] VS Code (version 1.72.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 2 categories.`

**Comment by huycozy** [comment]
@stoneLee81 Please see these StackOverflow answers if it can help: 
https://stackoverflow.com/questions/53394504/flutter-stuck-at-installing-build-app-outputs-apk-app-apk
https://stackoverflow.com/a/70892415/5921933

**Comment by stoneLee81** [comment]
@huycozy The problem seems to be that the path of master channel flutter compilation is wrong，The following is a comparison of the logs compiled by the two versions

flutter 3.3.9, it's okay.

> [   +2 ms] Stopping app 'app.apk' on GM1910.
[        ] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s a91bd13e shell am force-stop com.tomato.flutter_module.host
[ +157 ms] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s a91bd13e shell pm list packages com.tomato.flutter_module.host
[ +116 ms] package:com.tomato.flutter_module.host
[        ] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s a91bd13e shell cat /data/local/tmp/sky.com.tomato.flutter_module.host.sha1
[  +54 ms] 440440045e72e964d705164d9d41fa3609909589
[        ] Installing APK.
[   +1 ms] **Installing build/host/outputs/apk/app.apk...**
[        ] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s a91bd13e install -t -r /Users/haowmazs/data/SIP/flutter-webrtc-master/flutter_module/build/host/outputs/apk/app.apk
[+3370 ms] Performing Streamed Install
                    Success
[        ] Installing build/host/outputs/apk/app.apk... (completed in 3.4s)

flutter 3.7.0-0.0.pre, build error

> [+1422 ms] ✓  Built build/host/outputs/apk/debug/app-debug.apk.
[   +8 ms] Stopping app 'app-debug.apk' on SM T500.
[        ] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s R9TR301TCWJ shell am force-stop com.tomato.flutter_module.host
[  +83 ms] executing: /Users/haowmazs/Library/Android/sdk/platform-tools/adb -s R9TR301TCWJ shell pm list packages com.tomato.flutter_module.host
[ +135 ms] Installing APK.
[   +1 ms] **"build/host/outputs/apk/app-debug.apk" does not exist.**
[        ] Warning: Failed to install APK.
[        ] Error launching application on SM T500.
[+12738 ms] "flutter run" took 32,061ms.


the installing path that builds by flutter is different.

**Comment by darshankawar** [comment]
@stoneLee81 
What's the command are you runnnig to build the apk ?

Can you try the same with default counter app ? ie, create a new project on stable, switch to latest master and run the build apk command to see if it gives you same behavior or not.

**Comment by stoneLee81** [comment]
The problem seems to be in the compilation path. The apk has been compiled to the output/apk/debug directory. However, I copied the app-debug.apk file in the debug directory to the output/apk directory, and then compiled it once. it's successed.

**Comment by stoneLee81** [comment]
1 create project 
`flutter create testprj`
2 create module
`flutter create -t module --org com.example test`
3 open this module on vscode
4 build and run this module under android,error occurred
`✓  Built build/host/outputs/apk/debug/app-debug.apk.
"build/host/outputs/apk/app-debug.apk" does not exist.`
@darshankawar

**Comment by huycozy** [comment]
@stoneLee81 Thanks for providing `Steps to reproduce`.

I can see the issue on Flutter stable channel `3.7.0` but it seems to be fixed on Flutter master channel `3.7.0-28.0.pre.21`. Please retry and confirm this.

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.7.0, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (3 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Dolphin 2021.3.1.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio Dolphin 2021.3.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio Electric Eel 2022.1.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.7.0-28.0.pre.21, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.0-28.0.pre.21 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8d03af3424 (35 minutes ago), 2023-01-26 20:36:56 -0500
    • Engine revision 79c958fc7e
    • Dart version 3.0.0 (build 3.0.0-171.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Dolphin 2021.3.1.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio Dolphin 2021.3.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio Electric Eel 2022.1.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.

```
</details>

**Comment by PhilipSA** [comment]
I had the same issue and when i upgraded to `3.7.0-30.0.pre.7` it works again. Will there be a stable 3.7.1 version released or similar that fixes this issue?

**Comment by OnClickListener2048** [comment]
same problem

**Comment by huycozy** [comment]
>Will there be a stable 3.7.1 version released or similar that fixes this issue?

@PhilipSA It takes a couple of months to land stable fixes to release channels. You can read more about the build release channels here https://github.com/flutter/flutter/wiki/Flutter-build-release-channels.

@OnClickListener2048 Please take a look at my comment at https://github.com/flutter/flutter/issues/116173#issuecomment-1407985223 and retry on the latest master channel.

**Comment by burnermanx** [comment]
Same problem on Flutter 3.7.1.
Only worked with the workarround above

**Comment by bitmaxshenhe** [comment]
Flutter stable 3.7.1 3.7.0 都有这个问题

**Comment by xiaoxing1992** [comment]
Same problem on Flutter 3.7.1.
Only worked with above workarround

**Comment by xiaoxing1992** [comment]
> Flutter stable 3.7.1 3.7.0 都有这个问题

你是怎么解决的这个问题

**Comment by feicien** [comment]
Same problem on Flutter 3.7.2

**Comment by kevin-lemon** [comment]
Same problem on Flutter 3.7.2

**Comment by huycozy** [comment]
Hi guys, as mentioned in [above comment](https://github.com/flutter/flutter/issues/116173#issuecomment-1407985223), it seems to be fixed on the latest master channel, please try switching to master and then retry:

```
flutter channel master
flutter upgrade
```

**Comment by kevin-lemon** [comment]
> Hi guys, as mentioned in [above comment](https://github.com/flutter/flutter/issues/116173#issuecomment-1407985223), it seems to be fixed on the latest master channel, please try switching to master and then retry:
> 
> ```
> flutter channel master
> flutter upgrade
> ```

i tryed fixed，thanks

**Comment by kevin-lemon** [comment]
> Hi guys, as mentioned in [above comment](https://github.com/flutter/flutter/issues/116173#issuecomment-1407985223), it seems to be fixed on the latest master channel, please try switching to master and then retry:
> 
> ```
> flutter channel master
> flutter upgrade
> ```

i tryed fixed，thanks @huycozy

**Comment by huycozy** [comment]
> @huycozy channel stable how to fix？

It takes a couple of months to land stable fixes to release channels. You can read more about the build release channels here https://github.com/flutter/flutter/wiki/Flutter-build-release-channels.

I'll be closing this issue since it has already been fixed on the latest master channel. If someone else still experiences this issue on the latest **master** channel, kindly leave a comment below and I will reopen this. Thank you!

**Comment by burnermanx** [comment]
> Hi guys, as mentioned in [above comment](https://github.com/flutter/flutter/issues/116173#issuecomment-1407985223), it seems to be fixed on the latest master channel, please try switching to master and then retry:
> 
> ```
> flutter channel master
> flutter upgrade
> ```

But master channel is installing 3.8.x, not 3.7.x. It will fixed on a stable release of Flutter 3.7?

**Comment by roger66** [comment]
Same problem on Flutter 3.7.5.

**Comment by PhilipSA** [comment]
Yes this issue seem to have regressed in version `3.7.5`

**Comment by burnermanx** [comment]
Cherry-picking 5dabe102a2581a134dbf10ffc7fc8458d9bd1c05 fix this problem.

While a stable version doesn't released, it's a workarround cherrypicking this commit to your flutter sdk.

**Comment by antiv** [comment]
same problem with 3.7.6

**Comment by huycozy** [comment]
For anyone facing this issue, there was a CP for this issue: https://github.com/flutter/flutter/issues/121942 (filed by @burnermanx)

**Comment by carloscasanovad** [comment]
Does anyone has the step by step on how to fix the problem?

**Comment by burnermanx** [comment]
Fixed on 3.7.7

**Comment by wman0211** [comment]
same problem on 3.7.7
